### PR TITLE
fix(desktop): prepend https:// for bare hostnames in browser toolbar

### PR DESCRIPTION
**Pull request by NathanFant**
## Summary

- Typing `github.com` in the browser toolbar navigated to a relative path instead of `https://github.com`
- Extracts `displayUrl` and a new `normalizeUrl` utility from `BrowserToolbar.tsx` into a co-located `utils.ts`
- `normalizeUrl` detects inputs with no scheme and prepends `https://`, matching standard browser behavior
- URLs that already carry a scheme (`https://`, `http://`, `file://`, etc.) pass through unchanged

## Test plan

- [x] 15 unit tests added covering `displayUrl` and `normalizeUrl` — all pass (`bun test ...utils.test.ts`)
- [ ] Open a browser pane, type `github.com` and press Enter — should navigate to `https://github.com`
- [ ] Type `https://github.com` — should navigate normally, no double-scheme
- [ ] Type `http://localhost:3000` — should navigate normally
- [ ] Existing truncation behavior (long URLs, short URLs, edit mode) unaffected

---
## Summary by cubic
Fixes URL normalization in the browser toolbar and webview so bare hostnames open over https, while localhost and loopback IPs use http. Unifies normalization across form submit, autocomplete, and webview for consistent navigation.

- **Bug Fixes**
  - Prepend `https://` for bare hostnames and `hostname:port`; use `http://` for `localhost`, `127.0.0.1`, and `[::1]`.
  - Keep inputs with a scheme (`https`, `http`, `file`, `about:`, `data:`) unchanged; trim whitespace.
  - Parse bracketed IPv6 correctly; non-loopback addresses default to `https`; treat `github.com:3000` as host:port (not a scheme).
  - Preserve non-http schemes in the webview to avoid forcing `https` on `file://`, `about:`, and `data:` URLs.

- **Refactors**
  - Extract `displayUrl` and `normalizeUrl` into a utils module and replace the duplicate webview sanitizer; all navigation paths use `normalizeUrl`.
  - Add 24 unit tests covering blank pages, trailing slashes, scheme-only inputs, localhost/IPs, IPv6, hostname:port, and whitespace.

<sup>Written for commit e06485b570edbdfc44d10e7523f34822c44de3d8. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Navigation inputs are normalized before use for consistent link handling while displayed addresses remain user-friendly.
  * Bare hostnames now default to HTTPS; localhost-style hosts default to HTTP.
  * Displayed URLs omit unnecessary trailing slashes and treat special values like about:blank appropriately.

* **Tests**
  * Added comprehensive tests covering URL normalization and display across many edge cases.

### CODE REVIEW

File: apps/desktop/src/renderer/screens/main/components/WorkspaceView/ContentView/TabsContent/TabView/BrowserPane/components/BrowserToolbar/utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	return url.endsWith("/") ? url.slice(0, -1) : url;
+}
+
+const LOCALHOST_HOSTS = new Set(["localhost", "127.0.0.1", "[::1]"]);
+
+/** Extract the bare hostname from a schemeless input like `localhost:3000/path`. */
+function extractHost(input: string): string {
+	if (input.startsWith("[")) return "[::1]"; // IPv6
+	return input.split("/")[0].split(":")[0];
+}
```
[the remark this answers is not in the document]
Comment (NathanFant) [reply apps/desktop/src/renderer/screens/main/components/WorkspaceView/ContentView/TabsContent/TabView/BrowserPane/components/BrowserToolbar/utils.ts:21]: `d5977558` should resolve this.

**Comment by weltonrodrigo** [comment]
@NathanFant please make this support file:// urls because right now if i try to open `file:///tmp/report.html` it goes to `https://file///tmp/report.html`

**Comment by NathanFant** [comment]
Sure thing!

File: apps/desktop/src/renderer/screens/main/components/WorkspaceView/ContentView/TabsContent/TabView/BrowserPane/hooks/usePersistentWebview/usePersistentWebview.ts [hunk 2]
```diff
@@ -40,9 +40,14 @@ export function destroyPersistentWebview(paneId: string): void {
 // Helpers
```
[the remark this answers is not in the document]
Comment (NathanFant) [reply apps/desktop/src/renderer/screens/main/components/WorkspaceView/ContentView/TabsContent/TabView/BrowserPane/hooks/usePersistentWebview/usePersistentWebview.ts]: Duplication: sanitizeUrl in usePersistentWebview now deleted — replaced with the shared normalizeUrl import from BrowserToolbar/utils
hostname:port bug: github.com:3000 would have passed through sanitizeUrl unchanged (missing the dot-guard that normalizeUrl already has). Fixed by the deduplication above.
extractHost IPv6 claim: Not a real bug — extractHost parses the actual bracketed address and [::1] is only in LOCALHOST_HOSTS for the loopback check. Non-loopback IPv6 correctly gets https://.

**Comment by NathanFant** [comment]
Duplication: sanitizeUrl in usePersistentWebview now deleted — replaced with the shared normalizeUrl import from BrowserToolbar/utils
hostname:port bug: github.com:3000 would have passed through sanitizeUrl unchanged (missing the dot-guard that normalizeUrl already has). Fixed by the deduplication above.
extractHost IPv6 claim: Not a real bug — extractHost parses the actual bracketed address and [::1] is only in LOCALHOST_HOSTS for the loopback check. Non-loopback IPv6 correctly gets https://.

**Comment by NathanFant** [comment]
@cubic-dev-ai The search fallback in `sanitizeUrl` was never reachable in practice. `navigateTo` (the only consumer of the normalized URL at the webview level) is always called with output from `normalizeUrl` in `BrowserToolbar` — never with raw user input. `initialUrl` and history URLs are always proper URLs. The search fallback was dead code that created a false sense of safety while the real normalization happened upstream in the toolbar.

**Comment by NathanFant** [comment]
Hey @weltonrodrigo — this PR should fix the  issue you reported. Would you be able to test it out?
